pkg/trace/agent/run.go

```diff
@@ -78,17 +34,25 @@ func Run(ctx context.Context) {
 		panic(err)
 	}
 
+	if flags.Version {
+		fmt.Print(info.VersionString())
+		return
+	}
+
 	if flags.Info {
 		if err := info.Info(os.Stdout, cfg); err != nil {
-			os.Stdout.WriteString(fmt.Sprintf("failed to print info: %s\n", err))
-			os.Exit(1)
+			osutil.Exitf("failed to print info: %s\n", err)
 		}
 		return
 	}
 
-	// Exit if tracing is not enabled
+	if err := setupLogger(cfg); err != nil {
```

Comment:
You're now configuring the loader after `config.Load()` and `info.InitInfo()` get called - maybe losing some log lines. Are you sure you want to do that? Would you rather replace the logger after the configuration is loaded? Feel free to ignore if you think there's no impact here.

Reply:
I think we're good here as far as I'm concerned. If `config.Load` returns an error, it will have to fail startup and `info.InitInfo` isn't logging anything. It might be a bit premature to add a logger here to be future-proof, but I'm happy to hear other opinions from the APM team too. Happy to do it if we think we should.

Reply:
I think it's fair to leave logger init after those config loads, agree replacing the logger is a way to work around it, but what would an inconfigured logger do ? Print on stdout with an uncontrolled format ? This is what `osutil.Exitf()` will do anyway.

As a side note I'm curious why in some places we use `osutil.Exitf` and in others a `panic()` (eg: https://github.com/DataDog/datadog-agent/pull/3134/files#diff-d1648ec30e4f71df6861c250125facffR34)

Reply:
You're right. I missed that one. I've changed it to `osutil.Exitf`